src/dune/file_tree.ml

```diff
@@ -152,19 +161,72 @@ end = struct
       |> Result.ok
 end
 
+module Dirs_visited : sig
```

Comment:
I'm confused by this module's interface.

All functions return `t`, `Per_fn.t`, or `Per_fn.dirs_visited`, and all of those types are abstract. How does one extract information out of it? :-)

Reply:
I guess the interface is that `add` is failing if you add a duplicate. Might be worth adding a comment and renaming `add` to `add_exn`, but fair enough.